src/cli/peakdetector/PeakDetectorCLI.cpp

```diff
@@ -1070,6 +1070,52 @@ void PeakDetectorCLI::saveCSV(string setName) {
     }
     csvreports->closeFiles();
 
+	// Check whether there's a missing C12 PARENT label in the written CSV file.
+    bool atLeastOneIsotopeLabelIsEmpty = false;
+    bool fileNeedsCorrection = false;
+    vector<int> groupsWithMissingLabels;
+    CsvParser *csvParser = CsvParser_new(fileName.c_str(), ",", 1);
+    CsvRow *currentRow = NULL;
+    // Iterates over each row, continuously assigning a pointer to an unread CsvRow.
+    while (currentRow = CsvParser_getRow(csvParser)) {
+        // eighth field is the isotope label.
+        string isotopeLabel = string(currentRow->fields_[7]);
+        if (isotopeLabel.empty()) {
+            int groupId = atoi(currentRow->fields_[2]);
+            groupsWithMissingLabels.push_back(groupId);
+            atLeastOneIsotopeLabelIsEmpty = true;
+        } else {
+            if (atLeastOneIsotopeLabelIsEmpty)
+                fileNeedsCorrection = true;
+        }
+    }
+
+    if (fileNeedsCorrection) {
+        // rewrite file if needed
+        csvreports->openGroupReport(fileName);
+		csvreports->groupId = 0;
+
+		sort(groupsWithMissingLabels.begin(), groupsWithMissingLabels.end());
+        for (int i=0; i<mavenParameters->allgroups.size(); i++) {
+			PeakGroup &group = mavenParameters->allgroups[i];
+			if (binary_search(groupsWithMissingLabels.begin(), groupsWithMissingLabels.end(), csvreports->groupId + 1)) {
+				PeakGroup newGroup = group;
+				Compound *compound = group.getCompound();
+				float compoundMz = MassCalculator::computeMass(compound->formula, mavenParameters->getCharge(compound));
+				float cutoffDist = massCutoffDist(group.meanMz, compoundMz, mavenParameters->massCutoffMerge);
+		    	if (cutoffDist < mavenParameters->massCutoffMerge->getMassCutoff()) {
+					PeakGroup childGroup = group;
+					childGroup.tagString = "C12 PARENT";
+					newGroup.children.push_back(childGroup);
+					csvreports->addGroup(&newGroup);
```

Comment:
@saifulbkhan would suggest adding a comment here that groupId is updated in the addGroup function. Not necessary, just helps readability.

Reply:
Will add.